scripts/linters/html_linter.py

```diff
@@ -265,6 +267,64 @@ def all_filepaths(self):
         """Return all filepaths."""
         return self.html_filepaths
 
+    def _check_safe_pipe_usage(self):
+        """This function checks if the usage of safe is warranted or not."""
+
+        if self.verbose_mode_enabled:
+            python_utils.PRINT('Starting Safe Pipe checks')
+            python_utils.PRINT('----------------------------------------')
+
+        html_files_to_lint = self.html_filepaths
+        stdout = sys.stdout
+
+        failed = False
+        summary_messages = []
+
+        with linter_utils.redirect_stdout(stdout):
+            for filepath in html_files_to_lint:
+                file_content = self.file_cache.read(filepath)
+                if '| safe' in file_content or '|safe' in file_content:
```

Comment:
Check the file line by line and use a regex to match the content of that line. Also, print the line number where this will be found.

Reply:
This runs into one problem:

Currently the `WARRANTED_USAGE" has the following structure:
```
{
name_of_file: total_occurances
}
```

If the totalOccurances in the file don't match the specified amount, which line should I call out the error is in. Because I am looking at a cumulative figure, I won't be able to pinpoint a line.

Reply:
You can add a variable to count the number of occurrences, right?

Reply:
I did that here. https://github.com/oppia/oppia/pull/10290/files#diff-ff29624830fd49a18ea3ae12fbf4f537R287